inference-engine/src/transformations/src/transformations/common_optimizations/interpolate_sequence_fusion.cpp

```diff
@@ -0,0 +1,196 @@
+              [](const std::pair<int64_t, float>& a, const std::pair<int64_t, float>& b) {
+                  return a.first < b.first;
+              });
+    std::vector<int64_t> new_axes;
+    std::vector<float> new_scales;
+    for (const auto& as : axes_and_scales) {
+        new_axes.emplace_back(as.first);
+        new_scales.emplace_back(as.second);
+    }
+
+    auto new_scales_node = opset8::Constant::create(element::f32, {new_scales.size()}, new_scales);
+    auto new_axes_node = opset8::Constant::create(element::i64, {new_axes.size()}, new_axes);
+    auto shape_node = std::make_shared<opset8::ShapeOf>(fst->input_value(0));
+
+    auto gather_axis_node = opset8::Constant::create(element::i64, {1}, std::vector<int64_t>{0});
+    auto gather_node = std::make_shared<opset8::Gather>(shape_node, new_axes_node, gather_axis_node);
+    auto cast_shape_to_float = std::make_shared<opset8::Convert>(gather_node, element::f32);
+
+    auto mul_node = std::make_shared<opset8::Multiply>(cast_shape_to_float, new_scales_node);
+    auto eps_node = opset8::Constant::create(element::f32, {}, std::vector<float>{1.0e-5f});
+    auto add_node = std::make_shared<opset8::Multiply>(mul_node, eps_node);
+    auto floor_node = std::make_shared<opset8::Floor>(add_node);
+    auto cast_mul_result_to_int = std::make_shared<opset8::Convert>(floor_node, element::i64);
+
+    auto new_interpolate = fst->clone_with_new_inputs({fst->input_value(0), cast_mul_result_to_int, new_scales_node, new_axes_node});
+
+    return {new_scales_node, new_axes_node, shape_node, gather_axis_node, gather_node, cast_shape_to_float, mul_node, eps_node,
+            add_node, floor_node, cast_mul_result_to_int, new_interpolate};
+}
+} // namespace
+
+NGRAPH_RTTI_DEFINITION(ngraph::pass::InterpolateSequenceFusion, "InterpolateSequenceFusion", 0);
+
+ngraph::pass::InterpolateSequenceFusion::InterpolateSequenceFusion() {
+    MATCHER_SCOPE(InterpolateSequenceFusion);
+    auto interpolate_pattern = ngraph::pattern::wrap_type<ngraph::opset8::Interpolate>();
+    ngraph::matcher_pass_callback callback = [=](ngraph::pattern::Matcher& m) {
+        auto snd_interpolate = std::dynamic_pointer_cast<opset8::Interpolate>(m.get_match_root());
+        if (!snd_interpolate) return false;
+
+        auto fst_interpolate = std::dynamic_pointer_cast<opset8::Interpolate>(snd_interpolate->input_value(0).get_node_shared_ptr());
+        if (!fst_interpolate) return false;
+
+        if (!can_be_fused(fst_interpolate, snd_interpolate)) return false;
+
+        NodeVector new_subgraph;
+        if (fst_interpolate->get_attrs().shape_calculation_mode == ngraph::opset8::Interpolate::ShapeCalcMode::SIZES) {
+            new_subgraph = subgraph_for_sizes_calculation_mode(fst_interpolate, snd_interpolate);
+        } else {
+            new_subgraph = subgraph_for_scales_calculation_mode(fst_interpolate, snd_interpolate);
+        }
+
+        auto& new_interpolate = new_subgraph.back();
```

Comment:
You also need to call register_new_node so we can fuse sequence of > 2 Interpolates.
Also please add a test with sequence like Interp->Interp->Interp which will be fused into Interp.

Reply:
Such tests have already been written: see tests `InterpolateSequenceFusion5D1` and `InterpolateSequenceFusion5D2`.

Reply:
Added a call of `register_new_node`.